pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -506,6 +526,33 @@ def _binary_op_method_timedeltalike(op, name):
 # ----------------------------------------------------------------------
 # Timedelta Construction
 
+def _value_from_iso_match(match):
+    """
+    Extracts and cleanses the appropriate values from a match object with
+    groups for each component of an ISO 8601 duration
+
+    Parameters
+    ----------
+    match:
+        Regular expression with groups for each component of an ISO 8601
+        duration
+
+    Returns
+    -------
+    int
+        Precision in nanoseconds of matched ISO 8601 duration
+    """
+    match_dict = {k: v for k, v in match.groupdict().items() if v}
+    for comp in ['milliseconds', 'microseconds', 'nanoseconds']:
+        if comp in match_dict:
+            match_dict[comp] ='{:0<3}'.format(match_dict[comp])
+
+    match_dict = {k: int(v) for k, v in match_dict.items()}
+    nano = match_dict.pop('nanoseconds', 0)
+
+    return nano + convert_to_timedelta64(timedelta(**match_dict), 'ns')
```

Comment:
just return the nanoseconds here

Reply:
Not sure I follow - don't we need to the timedelta64 to get the nanosecond precision of all the components?

Reply:
better to use timedelta_from_spec and construct the number of nanoseconds, then simply return that